glean-core/android/src/main/java/mozilla/telemetry/glean/Glean.kt

```diff
@@ -162,8 +164,10 @@ open class GleanInternalAPI internal constructor () {
         @Suppress("EXPERIMENTAL_API_USAGE")
         Dispatchers.API.flushQueuedInitialTasks()
 
-        // At this point, all metrics and events can be recorded.
-        ProcessLifecycleOwner.get().lifecycle.addObserver(gleanLifecycleObserver)
+        MainScope().launch {
```

Comment:
**Copied from a-c**

By looking at the docs and the [rationale for this](https://github.com/Kotlin/kotlinx.coroutines/issues/829), looks like we should declare `private val mainScope = MainScope()` in the `GleanInternalAPI` class and use `mainScope.launch` here. This follows the same pattern we've used for `CoroutineScope` in the dispatcher.

I don't know if that's *strictly* required, but I wouldn't want this to have surprising behaviours due to its short lived existence.

Reply:
This is used just like this in many places throughout Android Components, it's one of the reasons I saw and suggested it.

Reply:
I'm really glad we discovered this interesting little tidbit about lifecycle observers early rather than later.

Reply:
Actually @Dexterp37 could you explain what exactly makes you worried that the MainScope() will be short lived?

Reply:
`MainScope` is a `CoroutineScope` and, as such, might do something on its setup and teardown. `MainScope`, used like that, probably lives long enough to just call `launch`. What happens to the children coroutines (the one triggered by `launch`), when `MainScope()` is GC'd? I don't know if that's a problem or not, but docs use it differently. Anyway, none it this matters as there's on-going discussion on the a-c side: this might need to change.

Reply:
Thanks!  Although I would hope that the scope returned by the `MainScope()` returned a scope that didn't go "out of scope", considering recent lessons learned about Android, it's probably better to be safe than sorry.